Without that, build fails with an undefined reference to fmod@GLIBC_2.38 in libxml2.so, on x86_64. Have no idea how could it happen.
diff -uNr a/src/meson.build b/src/meson.build
--- a/src/meson.build	2023-11-29 18:16:40.269713092 +0000
+++ b/src/meson.build	2023-11-29 18:17:04.333673708 +0000
@@ -29,7 +29,7 @@
 if get_option('scanner')
 	# wayland-scanner
 
-	scanner_deps = [ dependency('expat') ]
+	scanner_deps = [ dependency('expat'), cc.find_library('m') ]
 	scanner_args = [ '-include', 'config.h' ]
 
 	if get_option('dtd_validation')
